retis/src/core/probe/kernel/bpf/include/common.h

```diff
@@ -144,6 +144,74 @@ HOOK(9)
 /* Keep in sync with its Rust counterpart in crate::core::probe::kernel */
 #define HOOK_MAX 10
 
+__attribute__ ((noinline))
+int ctx_hook(struct retis_context *ctx)
+{
+	volatile int ret = 0;
+	if (!ctx)
+		return 0;
+	return ret;
+}
+
+#define DEFINE_CTX_HOOK(statements)						\
+	SEC("ext/hook")								\
+	int hook(struct retis_context *ctx)					\
+	{									\
+		if (!ctx)							\
+			return 0;						\
+		statements							\
+	}
+
+static __always_inline int extend_ctx_nft(struct retis_context *ctx)
+{
+	struct nft_traceinfo___6_3_0 *info_63;
+	const struct nft_pktinfo *pkt;
+	struct nft_traceinfo *info;
+
+	if (retis_arg_valid(ctx, sk_buff) ||
+	    !bpf_core_type_exists(struct nft_traceinfo) ||
+	    !bpf_core_type_exists(struct nft_pktinfo))
+		return 0;
+
+	info = retis_get_nft_traceinfo(ctx);
+	if (!info)
+		return 0;
+
+	info_63 = (struct nft_traceinfo___6_3_0 *)info;
+	if (bpf_core_field_exists(info_63->pkt))
+		pkt = BPF_CORE_READ(info_63, pkt);
+	else
+		pkt = retis_get_nft_pktinfo(ctx);
+
+	if (pkt) {
+		ctx->regs.reg[EXT_REG_SKB] = (u64)BPF_CORE_READ(pkt, skb);
+		ctx->offsets.sk_buff = EXT_REG_SKB;
```

Comment:
nit: we could provide this as an helper; would be easier for users of the ctx hook.